### ISSUE TITLE
ContextMenu for the map does not work properly

### ISSUE THREAD

**Issue by n0gare** [Leaflet/Leaflet#5194]
Hello,

I've searched through the issues to make sure it's not yet reported.
I am using Leaflet.contextmenu to attach a contextmenu to:
   -the map
   -the nodes (markers)
   -the edges (polylines)
This is how the contextmenu setup looks like for the map but its the same structure for nodes and edges.
        map = L.map('map', {
            contextmenu: true,
            contextmenuWidth: 140,
            contextmenuItems: [{
                text: 'Show coordinates',
                callback: showCoordinates
            }]);
I did not noticed this issue (maybe bug) at first.
The problem is that after I open the contextmenu of a node or an edge the contextmenu of the map will never show up again.
It seems like the contextmenu of the node and the edge overwrite the one of the map.

After I tried to find a solution myself for quite some time, my next step was to look for alternatives. 
So I used the Leaflet.Dialog Plugin to create my own contextmenu for the map. It worked pretty well,  the only problem is that when I open the contextmenu of an edge, the contextmenu of the map opens in additon to the regular contextmenu. 
It looks like the polyline is treated like the map.

To open the contextmenu of the map I wrote : 
        map.on("contextmenu", function f () {
            dialog.setContent(content)
            .addTo(map);
            dialog.open();
        });
I also increased the weight, to make sure I dont accidentally hit the map.

To sum up I have two questions.
1. Why is it not possible to use the Leaflet.Contextmenu for the map, polylines and markers?
2.Why does map.on("contextmenu") gets fired when the user right clicks a polyline?

I hope you understand what my problem is, if not I am happy to explain it in more detail.

Best regards!

**Comment by perliedman** [comment]
Hi there!

Regarding Leaflet.Contextmenu, you should file an issue with that plugin. It's not maintained by the Leaflet team, so we have no detailed insight as to what the problem might be.

Regarding your second question, events on path's (like polylines) bubble to their container, which is the map. This works much like other DOM events, where a click on a `div` will also bubble to its container unless you stop it (see [`stopPropagation`](http://leafletjs.com/reference-1.0.2.html#domevent-stoppropagation)).
